src/test/java/org/folio/dew/batch/bursarfeesfines/service/BursarTokenFormatterTest.java

```diff
@@ -0,0 +1,980 @@
+
+    dateType = BursarExportTokenDateType.QUARTER;
+    assertThat(
+      BursarTokenFormatter.processDateToken(
+        testDateTime,
+        dateType,
+        lengthControl
+      ),
+      is("1")
+    );
+
+    dateType = BursarExportTokenDateType.WEEK_OF_YEAR;
+    assertThat(
+      BursarTokenFormatter.processDateToken(
+        testDateTime,
+        dateType,
+        lengthControl
+      ),
+      is("5")
+    );
+
+    dateType = BursarExportTokenDateType.WEEK_OF_YEAR_ISO;
+    assertThat(
+      BursarTokenFormatter.processDateToken(
+        testDateTime,
+        dateType,
+        lengthControl
+      ),
+      is("5")
+    );
+
+    dateType = BursarExportTokenDateType.WEEK_YEAR;
+    assertThat(
+      BursarTokenFormatter.processDateToken(
+        testDateTime,
+        dateType,
+        lengthControl
+      ),
+      is("2023")
+    );
+
+    dateType = BursarExportTokenDateType.WEEK_YEAR_ISO;
+    assertThat(
+      BursarTokenFormatter.processDateToken(
+        testDateTime,
+        dateType,
+        lengthControl
+      ),
+      is("2023")
+    );
+  }
+
+  @Test
+  void testProcessAggregateToken() {
+    int testNumRows = 25;
+    BigDecimal testAmount = new BigDecimal("123.45");
+    BursarExportTokenAggregate aggregate = new BursarExportTokenAggregate();
+
+    aggregate.setValue(ValueEnum.NUM_ROWS);
+    assertThat(
+      BursarTokenFormatter.processAggregateToken(
+        aggregate,
+        testNumRows,
+        testAmount
+      ),
+      is("25")
+    );
+
+    aggregate.setValue(ValueEnum.TOTAL_AMOUNT);
+    aggregate.setDecimal(true);
+    assertThat(
+      BursarTokenFormatter.processAggregateToken(
+        aggregate,
+        testNumRows,
+        testAmount
+      ),
+      is("123.45")
+    );
+    aggregate.setDecimal(false);
+    assertThat(
+      BursarTokenFormatter.processAggregateToken(
+        aggregate,
+        testNumRows,
+        testAmount
+      ),
+      is("12345")
+    );
+  }
+
+  @Test
+  void testFormatHeaderFooterToken() {
+    int aggregateNumRows = 25;
+    BigDecimal aggregateAmount = new BigDecimal("123.45");
+
+    BursarExportTokenConstant constant = new BursarExportTokenConstant();
+    constant.setValue("test");
+    assertThat(
+      BursarTokenFormatter.formatHeaderFooterToken(
+        constant,
+        aggregateNumRows,
+        aggregateAmount
+      ),
+      is("test")
+    );
+
+    BursarExportTokenCurrentDate currentDate = new BursarExportTokenCurrentDate();
+    currentDate.setValue(BursarExportTokenDateType.YEAR_LONG);
+    currentDate.setTimezone("UTC");
+    String currentYear = String.valueOf(ZonedDateTime.now().getYear());
+    assertThat(
+      BursarTokenFormatter.formatHeaderFooterToken(
+        currentDate,
+        aggregateNumRows,
+        aggregateAmount
+      ),
+      is(currentYear)
+    );
+
+    BursarExportTokenAggregate aggregate = new BursarExportTokenAggregate();
+    aggregate.setValue(ValueEnum.NUM_ROWS);
+    assertThat(
```

Comment:
This is a similar case to the Enums as `BursarExportHeaderFooter` can only be one of the three `BursarExportTokenConstant`,  `BursarExportTokenCurrentDate`, `BursarExportTokenAggregate`, which are already tested by the if/else if statements. If I try passing in a different token, it will just throw an error.